Boards 7.3 and doc reorg (#5881)
mattermost/docs: source/boards/share-and-collaborate.rst

Share and collaborate
=====================

.. include:: ../_static/badges/allplans-cloud-selfhosted.rst
  :start-after: :nosearch:

Board permissions
-----------------

From Mattermost Boards v7.2, boards belong to teams, and any member of that team can be granted access to a board. Prior to Mattermost Boards v7.2, boards are tied to channel workspaces, and board membership is determined by channel membership.

Roles
-----

> The level of access to a board is determined by a user’s assigned board role.

Reviewer: Decided against using the table?

Author: Still adding the table. :)

- **Admin:** Can modify the board, its contents, and its permissions. By default, board creators are also admins of the board.
- **Editor:** Can modify the board and its contents.

Manage team access
~~~~~~~~~~~~~~~~~~~

> Board admins can manage team access to their board by selecting **Share** in the top-right corner of the board. On the dropdown next to **Everyone at… Team** option, select a minimum board role for everyone on the team. 

Reviewer: ```suggestion
Board admins can manage team access to their board by selecting **Share** in the top-right corner of the board. On the dropdown next to **Everyone at… Team** option, select a default board role for everyone on the team. 
```

> Depending on the role selected, everyone on the team will have access to the board with a minimum of the permissions from the role selected. Users can get elevated permissions based on their individual board membership. The default team access for a newly created board is **None**, which means nobody on the team has access to the board.

Reviewer: ```suggestion
Depending on the role selected, everyone on the team will have access to the board with the permissions from the role selected. Users can get elevated permissions based on their individual board membership. The default team access for a newly created board is **None**, which means nobody on the team has access to the board.
```

Manage individual board membership
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Only board admins manage user permissions on a board, including adding, changing, and removing members.

Reviewer: ```suggestion
Only board admins can manage user permissions on a board, including adding, changing, and removing members.
```

> To add individual users from the team as explicit members of the board, open the Share dialog on the board, search for individual team members, then assign a role to set their permissions for the board. The role for individual board members overrides any role specified for team access.

Reviewer: ```suggestion
To add individual users from the team as explicit members of the board, open the **Share** dialog on the board, search for individual team members, then assign a role to set their permissions for the board. The role for individual board members overrides any role specified for team access.
```

- To change a board member’s role, open the **Share** dialog, select the role dropdown next to the user’s name, then select another role from the list.

> - To remove a member from a board, open the **Share** dialog, select the role dropdown next to the user’s name, then select Remove member.

Reviewer: ```suggestion
- To remove a member from a board, open the **Share** dialog, select the role dropdown next to the user’s name, then select **Remove member**.
```

Channel role groups
~~~~~~~~~~~~~~~~~~~

> Board admins can add a channel to a board to grant all its members Editor access. To do this, select **Share** in the top-right corner of the board, search for the channel name, and add it to the board as a user. The default role is Editor. Doing so also links the board back to the channel, where the board will appear on the channel RHS.

Reviewer: ```suggestion
Board admins can add a channel to a board to grant all its members Editor access. To do this, select **Share** in the top-right corner of the board, search for the channel name, and add it to the board as a user. The default role is Editor. Doing so also [links the board back to the channel](ADD URL to "Link a board to a channel"), where the board will appear on the channel RHS.
```

> To unlink the channel from the board, open the **Share** dialog, select the role dropdown next to the channel’s name, then select Unlink.

Reviewer: ```suggestion
To unlink the channel from the board, open the **Share** dialog, select the role dropdown next to the channel’s name, then select **Unlink**.
```

.. note:: 
  
  A board can only be linked to one channel at a time. Linking another channel to the same board will automatically remove the link from the previous channel.

Share a board
--------------

Boards can be shared internally with your team or published externally with limited accessibility.

Share a board internally
~~~~~~~~~~~~~~~~~~~~~~~~

To share a board with team members internally, select **Share** in the top-right corner of the board, then select **Copy link** from the **Share** tab below. Paste the copied link in a channel or direct message to share the board with other team members. Only team members who have permissions to the board will be able to open the board from the shared link.

Share a board publicly
~~~~~~~~~~~~~~~~~~~~~~

> Sharing boards publicly is disabled by default. This means that the Publish tab is not available from the Share dialog. To enable public board sharing:

Reviewer: ```suggestion
Sharing boards publicly is disabled by default. This means that the **Publish** tab is not available from the **Share** dialog. To enable public board sharing:
```

4. Select **Copy link**.

Paste the link anywhere you want to share the board. Anyone with the link will be able to view the board, but they won’t be able to edit the board.

Select the **Regenerate Token** icon in the URL box if you want to invalidate all the previously shared links. Confirm the action to regenerate the token.

Share cards on Channels
-----------------------

Cards can be linked and shared with team members directly on Mattermost Channels. When you share a link to a card within Channels, the card details are automatically displayed in a preview. This preview highlights what the card is about at a glance without having to navigate to it.

To share a card, you'll need to copy the card link first:

- Open a card and select the options menu **(...)** at the top right of the card, then select **Copy link**.
- Alternatively, you can open the board view and hover your mouse over any card to access the options menu **(...)** for the card and select **Copy link** from there.

After you've copied the link, paste it into any channel or direct message to share the card. A preview of the card will display within the channel with a link back to the card on Boards.

> Import and export a board

Reviewer: @justinegeffen This section should be moved to the "Migrate boards" section. There's also updated content for that as well that should be added there

Depending on the role selected, everyone on the team will have access to the board with the permissions from the role selected. Users can get elevated permissions based on their individual board membership. The default team access for a newly created board is **None**, which means nobody on the team has access to the board.

Manage individual board membership
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Only board admins can manage user permissions on a board, including adding, changing, and removing members.

To add individual users from the team as explicit members of the board, open the **Share** dialog on the board, search for individual team members, then assign a role to set their permissions for the board. The role for individual board members overrides any role specified for team access.

- To change a board member’s role, open the **Share** dialog, select the role dropdown next to the user’s name, then select another role from the list.
- To remove a member from a board, open the **Share** dialog, select the role dropdown next to the user’s name, then select **Remove member**.

Channel role groups
~~~~~~~~~~~~~~~~~~~

> Board admins can add a channel to a board to grant all its members Editor access. To do this, select **Share** in the top-right corner of the board, search for the channel name, and add it to the board as a user. The default role is Editor. Doing so also [links the board back to the channel](ADD URL to "Link a board to a channel"), where the board will appear on the channel RHS.

Author: ```suggestion
Board admins can add a channel to a board to grant all its members Editor access. To do this, select **Share** in the top-right corner of the board, search for the channel name, and add it to the board as a user. The default role is Editor. Doing so also `links the board back to the channel <https://docs.mattermost.com/boards/navigate-boards.html#link-a-board-to-a-channel>`_, where the board will appear on the channel RHS.
```
